core/src/main/java/org/apache/iceberg/encryption/AesGcmOutputStream.java

```diff
@@ -0,0 +1,138 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.encryption;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import org.apache.iceberg.io.PositionOutputStream;
+
+public class AesGcmOutputStream extends PositionOutputStream {
+
+  private final Ciphers.AesGcmEncryptor gcmEncryptor;
+  private final PositionOutputStream targetStream;
+  private final byte[] fileAadPrefix;
+  private final byte[] singleByte;
+
+  private byte[] plainBlock;
+  private byte[] cipherBlock;
+  private int positionInPlainBlock;
+  private long streamPosition;
+  private int currentBlockIndex;
+  private boolean isHeaderWritten;
+
+  AesGcmOutputStream(PositionOutputStream targetStream, byte[] aesKey, byte[] fileAadPrefix) {
+    this.targetStream = targetStream;
+    this.gcmEncryptor = new Ciphers.AesGcmEncryptor(aesKey);
+    this.plainBlock = new byte[Ciphers.PLAIN_BLOCK_SIZE];
+    this.cipherBlock = new byte[Ciphers.CIPHER_BLOCK_SIZE];
+    this.positionInPlainBlock = 0;
+    this.streamPosition = 0;
+    this.currentBlockIndex = 0;
+    this.fileAadPrefix = fileAadPrefix;
+    this.isHeaderWritten = false;
+    this.singleByte = new byte[1];
+  }
+
+  @Override
+  public void write(int b) throws IOException {
+    singleByte[0] = (byte) (b & 0x000000FF);
+    write(singleByte);
+  }
+
+  @Override
+  public void write(byte[] b, int off, int len) throws IOException {
+    if (!isHeaderWritten) {
+      writeHeader();
+    }
+
+    if (b.length - off < len) {
+      throw new IOException(
+          "Insufficient bytes in buffer: " + b.length + " - " + off + " < " + len);
+    }
+
+    int remaining = len;
+    int offset = off;
+
+    while (remaining > 0) {
+      int freeBlockBytes = plainBlock.length - positionInPlainBlock;
+      int toWrite = Math.min(freeBlockBytes, remaining);
+
+      System.arraycopy(b, offset, plainBlock, positionInPlainBlock, toWrite);
+      positionInPlainBlock += toWrite;
+      offset += toWrite;
+      remaining -= toWrite;
+
+      if (positionInPlainBlock == Ciphers.PLAIN_BLOCK_SIZE) {
+        encryptAndWriteBlock();
+      }
+    }
+
+    streamPosition += len;
+  }
+
+  @Override
+  public long getPos() throws IOException {
+    return streamPosition;
+  }
+
+  @Override
+  public void flush() throws IOException {
+    targetStream.flush();
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (!isHeaderWritten) {
+      writeHeader();
+    }
+
+    if (positionInPlainBlock > 0) {
+      encryptAndWriteBlock();
+    }
+
+    targetStream.close();
+    plainBlock = null;
+    cipherBlock = null;
+  }
+
+  private void writeHeader() throws IOException {
+    byte[] headerBytes =
+        ByteBuffer.allocate(Ciphers.GCM_STREAM_HEADER_LENGTH)
+            .order(ByteOrder.LITTLE_ENDIAN)
+            .put(Ciphers.GCM_STREAM_MAGIC_ARRAY)
+            .putInt(Ciphers.PLAIN_BLOCK_SIZE)
+            .array();
```

Comment:
This byte array is made entirely from constants. Can you make it a constant byte array?